File: src/discord/commands/obama.ts
import { AttachmentBuilder, CommandInteraction } from "discord.js";
import { SlashCommandBuilder } from "discord.js";
import { readFileSync } from "node:fs";

const buffer = readFileSync("images/obamacomputer.jpg");
module.exports = {
	data: new SlashCommandBuilder().setName("obama").setDescription("obama"),

	async execute(interaction: CommandInteraction) {
		await interaction.reply({ files: [new AttachmentBuilder(buffer, { name: "obamacomputer.png", description: "obama plays osu!" })] });
	},
};

File: src/discord/commands/obama.cs
using System.Reflection;
using Discord;
using Discord.Interactions;

public partial class DiscordCommands : InteractionModuleBase
{
    [SlashCommand("obama", "obama plays osu!")]
    public async Task obama()
    {
        FileAttachment file = new FileAttachment(Settings.PATH_OBAMA);
        file.Description = "obama plays osu!";

        await RespondWithFileAsync(file);
    }
}